### PR TITLE
Introduction to Secrets Management with Doppler

### LINKED ISSUE #5943: [Security] Introduction to Secrets Management with Doppler
### Topic Suggestion
[Security] Introduction to Secrets Management with Doppler

#### Proposed article introduction
Without the right tool and knowledge, developers often fail to keep their secrets safe. But what are secrets? You may ask. 3rd party API Keys, database configs and urls, passwords and other data alike form secrets that developers are needed to safeguard in their applications.
In this article, we will look at how to manage developer secrets the best way possible, using Doppler solution.
#### Key takeaways 
1. What are Secrets
2.  Why manage Secrets
3.  Traditional and Unconventional ways of Managing Secrets
4.  Introducing Doppler- Managing Secrets the right way.
5.  Setting up Doppler CLI/Doppler Dashboard
6. Conclusion

#### Article quality
This article will be more user based. The reader will be shown how to start a project, add secrets, delete secrets setup environments and more. Besides a learner will obtain advise on some unconventional ways of managing secrets and then learn the best approach required when working with developer secrets.

#### References
N/A

### PR DESCRIPTION
**Pull request by LynetteMwe**
> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This fixes #5943  

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @LynetteMwe ,

Welcome to EngEd community!

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started with the review process.

Meanwhile, we suggest you to go through other approved PRs/articles to see how the authors have formatted their articles, to avoid any potential closure of your PR.

Thanks.

**Comment by LynetteMwe** [comment]
> Hello @LynetteMwe ,
> 
> Welcome to EngEd community!
> 
> Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started with the review process.
> 
> Meanwhile, we suggest you to go through other approved PRs/articles to see how the authors have formatted their articles, to avoid any potential closure of your PR.
> 
> Thanks.

Noted with thanks, very eager to get this reviewed.

File: content/articles/introduction-to-secrets-management-with-doppler/index.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+Doppler supports seamless integrations for multiple cloud vendors and platforms. 
+This platform also provides dashboard-based integration of projects with other secrets managers such as AWS Secrets Manager, Parameter store and Hashicorp Vault.
+
+Secrets in Doppler can also be synchronized to external secrets stores, saving developers time to focus on building products and features.
+
+
+### Doppler CLI/Doppler Dashboard
+
+[Doppler CLI](https://docs.doppler.com/docs/cli) is a lightweight installable file that runs in most operating systems. Installing the binary file require package managers such as scoop.
+Doppler CLI is convenient for local development, CI/CD pipelines and production environments.
```
Comment (srishilesh) [inline content/articles/introduction-to-secrets-management-with-doppler/index.md:99]: How?

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @LynetteMwe ,

Good job on your first article.

I've suggested some major changes for improving the quality of the article. Make sure to go through all the suggestions carefully and address them.

I would also highly recommend you to go through other approved PRs/published articles to see if you have missed out any. This would help avoid PR closure.

Make sure your hero image is of 600x400 dimension, and you perform a grammar check.

Ping me when done.

Thanks.

**Comment by lalith1403** [comment]
PFA the plagiarism report. It is higher than we typically accept. Please make the changes. Thank you
[Introduction to Secrets Management with Doppler.pdf](https://github.com/section-engineering-education/engineering-education/files/7939728/Introduction.to.Secrets.Management.with.Doppler.pdf)
.

**Comment by LynetteMwe** [comment]
> PFA the plagiarism report. It is higher than we typically accept. Please make the changes. Thank you [Introduction to Secrets Management with Doppler.pdf](https://github.com/section-engineering-education/engineering-education/files/7939728/Introduction.to.Secrets.Management.with.Doppler.pdf) .

Hello @lalith1403 , I have made adjustments.

**Comment by LynetteMwe** [comment]
> Hey @LynetteMwe ,
> 
> Good job on your first article.
> 
> I've suggested some major changes for improving the quality of the article. Make sure to go through all the suggestions carefully and address them.
> 
> I would also highly recommend you to go through other approved PRs/published articles to see if you have missed out any. This would help avoid PR closure.
> 
> Make sure your hero image is of 600x400 dimension, and you perform a grammar check.
> 
> Ping me when done.
> 
> Thanks.

I have completed the suggested revisions. I also had updated and uploaded front matter, image , and hero as required . Kindly confirm.

**Comment by srishilesh** [comment]
I'll complete by tech review by EOD

**Comment by LynetteMwe** [comment]
> I'll complete by tech review by EOD

Could you please adjust the other labels, like the hero and author profile.

**Comment by srishilesh** [comment]
> > I'll complete by tech review by EOD
> 
> Could you please adjust the other labels, like the hero and author profile.

Don't worry about it. I'm aware of that. On verifying them, I'll remove it.

**Comment by LynetteMwe** [comment]
> > > I'll complete by tech review by EOD
> > 
> > 
> > Could you please adjust the other labels, like the hero and author profile.
> 
> Don't worry about it. I'm aware of that. On verifying them, I'll remove it.

**Comment by srishilesh** [comment]
@LynetteMwe 

Why did you close the PR?

The tech review is in progress.

File: content/articles/introduction-to-secrets-management-with-doppler/index.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+doppler setup 
+``` 
+
+![Set Up](/engineering-education/introduction-to-secrets-management-with-doppler/local_configs.png)
+
+After connecting your local project with the Doppler project, you can access, filter, and download secret(s) as either plain values or JSON files.
+
+Run the command `doppler secrets get DOPPLER_PROJECT DOPPLER_CONFIG --plain` to select your current project and its environment, as shown below:
+
+![Project](/engineering-education/introduction-to-secrets-management-with-doppler/get.png)
```
Comment (srishilesh) [inline content/articles/introduction-to-secrets-management-with-doppler/index.md:149]: We don't have this image. Please add it. Make sure the extensions are in lowercase

**Review by srishilesh — APPROVED** [review]
@LynetteMwe 

Great article. It was very beginner friendly and easy to read.

I've made some minor changes.

Also, looks like you've missed out to add 2 images. Please ping me after you add them, I'll move it for the final review.

Thanks.

**Comment by LynetteMwe** [comment]
Hello @srishilesh , I am done with the images. Please confirm.

**Comment by LynetteMwe** [comment]
> @LynetteMwe
> 
> Great article. It was very beginner friendly and easy to read.
> 
> I've made some minor changes.
> 
> Also, looks like you've missed out to add 2 images. Please ping me after you add them, I'll move it for the final review.
> 
> Thanks.

Done

**Comment by ahmadmardeni1** [comment]
Great work team. @srishilesh @LynetteMwe 

Welcome to Section EngEd and congrats on your first article. @LynetteMwe

**Pull request merged**